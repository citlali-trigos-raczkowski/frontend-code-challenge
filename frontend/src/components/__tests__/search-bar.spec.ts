import { shallowMount, createLocalVue } from "@vue/test-utils";
import SearchBar from "../nav-components/search-bar.vue";
import VueRouter from "vue-router";
import { SearchBarMockData } from "../../../tests/mocks/mock-props";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("component/search-bar.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(SearchBar, {
      propsData: SearchBarMockData,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("placeholder displays as expected", () => {
    const wrapper = shallowMount(SearchBar, {
      propsData: SearchBarMockData,
      localVue,
    });
    const placeholder = wrapper.find("v-text-field-stub");
    expect(placeholder.attributes().placeholder).toEqual(
      "Search for your favorite Pokemon..."
    );
  });
});
