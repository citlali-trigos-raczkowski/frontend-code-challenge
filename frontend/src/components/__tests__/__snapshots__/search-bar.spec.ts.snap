// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/search-bar.vue renders correctly 1`] = `
<v-text-field-stub
  backgroundcolor=""
  clearable="true"
  clearicon="$clear"
  dark="true"
  data-cy="search-bar"
  dense="true"
  errorcount="1"
  errormessages=""
  filled="true"
  id="search-bar"
  loaderheight="2"
  messages=""
  placeholder="Search for your favorite Pokemon..."
  rounded="true"
  rules=""
  successmessages=""
  type="text"
  value=""
/>
`;
